ibm/resource_ibm_securitygroup_rule.go

```diff
@@ -0,0 +1,253 @@
+package ibm
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/softlayer/softlayer-go/datatypes"
+	"github.com/softlayer/softlayer-go/filter"
+	"github.com/softlayer/softlayer-go/services"
+	"github.com/softlayer/softlayer-go/sl"
+)
+
+func resourceIBMSecurityGroupRule() *schema.Resource {
+	return &schema.Resource{
+		Create:   resourceIBMSecurityGroupRuleCreate,
+		Read:     resourceIBMSecurityGroupRuleRead,
+		Delete:   resourceIBMSecurityGroupRuleDelete,
+		Exists:   resourceIBMSecurityGroupRuleExists,
+		Importer: &schema.ResourceImporter{},
+
+		Schema: map[string]*schema.Schema{
+			"direction": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				Description:  "Direction of rule: ingress or egress",
+				ValidateFunc: validateSecurityRuleDirection,
+			},
+			"ether_type": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "IP version IPv4 or IPv6",
+				Default:      "IPv4",
+				ForceNew:     true,
+				ValidateFunc: validateSecurityRuleEtherType,
+			},
+			"port_range_min": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+			},
+			"port_range_max": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+			},
+			"protocol": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "icmp, tcp or udp",
+				ForceNew:     true,
+				ValidateFunc: validateSecurityRuleProtocol,
+			},
+			"security_group_id": {
+				Type:     schema.TypeInt,
+				Required: true,
+				ForceNew: true,
+			},
```

Comment:
You're missing the field remote_ip (http://sldn.softlayer.com/reference/datatypes/SoftLayer_Network_SecurityGroup_Rule)
For verification, it can either be a CIDR or a single IP address.